### ISSUE TITLE
Function generic problem in 4.2

### ISSUE THREAD

**Issue by NipponSunrise** [HaxeFoundation/haxe#10218]
This code became invalid since `4.2`. In previous versions works as expected.

```haxe
class CreateModelCommand extends SimpleCommand {
    override public function execute(notification:INotification):Void {
        _registerProxy(new GameProxy(), IGameProxy);
    }

    private function _registerProxy<T:BaseProxyClass & I, I>(proxyImpl:T, interfaceClass:Class<I>):Void {
        ...
    }
```
where class `GameProxy` extends `BaseProxyClass`

**Error in `4.2.1`:**

`Class<app.model.game.IGameProxy> should be Class<Unknown<0> : app.model.game.GameProxy>
For function argument 'interfaceClass'`

**Comment by haxiomic** [comment]
There's some unknown details here; could you create a reproducible example zip or reproduce on try.haxe.org?

**Comment by NipponSunrise** [comment]
> 
> 
> There's some unknown details here: could you create a reproducible example zip or reproduce on try.haxe.org?

`4.2.1` https://try.haxe.org/#8dFFAA99
`4.1.5` https://try.haxe.org/#6a18c124

**Comment by haxiomic** [comment]
Perfect, thank you :)

**Comment by RealyUniqueName** [comment]
This should be fixed by aecf036a4fb2dea0a995b911462ed5c08efb238e (#10233)
Could you please check if the latest nightly build works for you? https://build.haxe.org/builds/haxe/

**Comment by RealyUniqueName** [comment]
Fixed by aecf036a4fb2dea0a995b911462ed5c08efb238e
